connect/runtime/src/test/java/org/apache/kafka/connect/util/RetryUtilTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.concurrent.Callable;
+
+@RunWith(PowerMockRunner.class)
+public class RetryUtilTest {
+
+    @Mock
+    private Callable<String> mockCallable;
+
+    @SuppressWarnings("unchecked")
+    @Before
+    public void setUp() throws Exception {
+        mockCallable = Mockito.mock(Callable.class);
+    }
+
+    @Test
+    public void TestSuccess() throws Exception {
+        Mockito.when(mockCallable.call()).thenReturn("success");
+        //EasyMock.expect(mockCallable.call()).andReturn("success").times(1);
+        try {
+            assertEquals("success", RetryUtil.retry(mockCallable, 10, 100));
+        } catch (Exception e) {
+            fail("Not expecting an exception: ", e.getCause());
+        }
+        Mockito.verify(mockCallable, Mockito.times(1)).call();
+    }
+
+    @Test
+    public void TestFailingRetries() throws Exception {
+        Mockito.when(mockCallable.call()).thenThrow(new TimeoutException());
+        try {
+            RetryUtil.retry(mockCallable, 10, 100);
+            fail("Expect exception being thrown here");
+        } catch (ConnectException e) {
+            // expecting a connect exception
+        } catch (Exception e) {
+            fail("Only expecting ConnectException");
+        }
+        Mockito.verify(mockCallable, Mockito.times(10)).call();
+    }
+
+    @Test
+    public void RetriesEventuallySucceed() throws Exception {
+        Mockito.when(mockCallable.call())
+                .thenThrow(new TimeoutException())
+                .thenThrow(new TimeoutException())
+                .thenThrow(new TimeoutException())
+                .thenReturn("success");
+        try {
+            assertEquals("success", RetryUtil.retry(mockCallable, 10, 100));
+        } catch (Exception e) {
+            fail("Not expecting an exception: ", e.getCause());
+        }
+        Mockito.verify(mockCallable, Mockito.times(4)).call();
+    }
+
+    @Test
+    public void FailWithNonRetriableException() throws Exception {
+        Mockito.when(mockCallable.call())
+                .thenThrow(new TimeoutException("timeout"))
+                .thenThrow(new TimeoutException("timeout"))
+                .thenThrow(new TimeoutException("timeout"))
+                .thenThrow(new TimeoutException("timeout"))
+                .thenThrow(new TimeoutException("timeout"))
+                .thenThrow(new NullPointerException("Non retriable"));
+        try {
+            for (int i = 0; i < 10; i++) {
+                RetryUtil.retry(mockCallable, 10, 100);
+            }
+            fail("Not expecting an exception: ");
+        } catch (ConnectException e) {
+            fail("Should fail with NPE");
+        } catch (NullPointerException e) {
+            // good
+        }
```

Comment:
What is the purpose of the for loop? Why not just:
```suggestion
        NullPointerException e = assertThrows(NullPointerException.class, RetryUtil.retry(mockCallable, 10, 100));
        assertTrue(e.getMesssage().startsWith("Non retriable"));
```